deployment/clouddeploy/deployer.go

```diff
@@ -357,18 +468,161 @@ func (p *Deployer) buildModifySpecs(
-				return nil, errors.New("all node groups must have the same version")
-			}
+			return nil, errors.New("invalid cloud provider for region selection")
+		}
+	}
+
+	deploymentProvider := ""
+	clusterProvider := ""
+	if cloudProvider == "aws" {
+		deploymentProvider = "aws"
+		clusterProvider = "hostedAWS"
+	} else if cloudProvider == "gcp" {
+		deploymentProvider = "gcp"
+		clusterProvider = "gcp"
+	} else if cloudProvider == "azure" {
+		deploymentProvider = "azure"
+		clusterProvider = "hostedAzure"
+	} else {
+		return nil, errors.New("invalid cloud provider for setup info")
+	}
+
+	p.logger.Debug("fetching deployment options project")
+
+	deploymentOpts, err := p.client.GetProviderDeploymentOptions(ctx, p.tenantID, &capellacontrol.GetProviderDeploymentOptionsRequest{
+		Provider: deploymentProvider,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get deployment options")
+	}
+
+	if clusterVersion == "" {
+		clusterVersion = deploymentOpts.ServerVersions.DefaultVersion
+	}
+	if clusterCidr == "" {
+		clusterCidr = deploymentOpts.SuggestedCidr
+	}
+
+	p.logger.Debug("creating a new cloud cluster")
+
+	clusterName := fmt.Sprintf("cbdc2_%s", clusterID)
+
+	specs, err := p.buildDeploySpecs(
+		ctx,
+		cloudProvider,
+		def.NodeGroups)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to build cluster specs")
+	}
+
+	createReq := &capellacontrol.DeployClusterRequest{
+		CIDR:        clusterCidr,
+		Description: "",
+		Name:        clusterName,
+		Package:     "developerPro",
+		ProjectId:   cloudProjectID,
+		TenantId:    p.tenantID,
+		Provider:    clusterProvider,
+		Region:      cloudRegion,
+		Override: capellacontrol.CreateOverrideRequest{
+			Image:  serverImage,
+			Server: clusterVersion,
+			Token:  p.overrideToken,
+		},
+		Server:   clusterVersion,
+		SingleAZ: false,
+		Specs:    specs,
+		Timezone: "PT",
+	}
+
+	p.logger.Debug("creating cluster", zap.Any("req", createReq))
+
+	newCluster, err := p.client.DeployCluster(ctx, p.tenantID, createReq)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create cluster")
+	}
+
+	cloudClusterID := newCluster.Id
+
+	p.logger.Debug("waiting for cluster creation to complete")
+
+	err = p.mgr.WaitForClusterState(ctx, p.tenantID, cloudClusterID, "healthy")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to wait for cluster deployment")
+	}
+
+	// we cheat for now...
+	clusters, err := p.ListClusters(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list clusters")
+	}
+
+	var thisCluster *ClusterInfo
+	for _, cluster := range clusters {
+		cluster := cluster.(*ClusterInfo)
+
+		if cluster.ClusterID == clusterID.String() {
+			thisCluster = cluster
 		}
 	}
+	if thisCluster == nil {
+		return nil, errors.New("failed to find new cluster after deployment")
+	}
 
+	return thisCluster, nil
+}
+
+func (p *Deployer) CreateNewCluster(ctx context.Context, def *clusterdef.Cluster, clusterVersion string) (deployment.ClusterInfo, error) {
```

Comment:
This should probably be private rather than public I would think?

Reply:
sure